### PR TITLE
Edge development

### PR DESCRIPTION
**Pull request by pitr-ch**
Combines #273 #272 #268 together, please feel free to comment there. I'll be pulling any updates made there to this PR.

- [x] Add Concurrent.executor as an easy access point to global executors
- [x] adapt Promise unification and publish in edge
- [x] actor integration
- [ ] Promise unification documentation
- [ ] merge actor updates #180

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,11 +1,13 @@
 source 'https://rubygems.org'
 
 gemspec name: 'concurrent-ruby'
+gemspec name: 'concurrent-ruby-edge'
```
Comment (jdantonio) [inline Gemfile]: When I was created the extension gem I ran into some problem with this. I can't remember what it was. That's why the extension gem isn't listed here. We'll need to take a look at this after we merge. Whatever it was may no longer be an issue.

**Comment by jdantonio** [comment]
This is a *very big* merge so it's hard to digest it all. Having reviewed all the pull requests it aggregates it seems good to me. I know you are very diligent and thorough so I'm sure it's ready. I say merge it!

I'll move `Exchanger` to the edge gem once we merge this. I'll rebase #270 and revisit it then, too.

**Comment by pitr-ch** [comment]
Thanks for the trust. I need to make few more actualizations and then I merge it.

**Comment by pitr-ch** [comment]
Leaving: 'Promise unification documentation', and 'merge actor updates #180' for another PR and merging.

**Pull request merged**